common/src/types/v0/transport/replica.rs

```diff
@@ -21,6 +22,103 @@ impl GetReplicas {
     }
 }
 
+/// Pool on which the resides
+#[derive(Serialize, Default, Debug, Clone, Eq, PartialEq)]
+pub struct ReplicaPool {
+    /// name of the pool
+    pub name: PoolId,
+    /// uuid of the pool
+    pub uuid: Option<PoolUuid>,
+}
+
+impl ReplicaPool {
+    /// Return a new `Self` from the provided arguments
+    pub fn new(name: &PoolId, uuid: Option<PoolUuid>) -> Self {
+        Self {
+            name: name.clone(),
+            uuid,
+        }
+    }
+    /// get the uuid of the pool
+    pub fn get_uuid(&self) -> Result<PoolUuid, uuid::Error> {
+        match self.uuid.clone() {
+            Some(uuid) => Ok(uuid),
+            None => {
+                // TODO add a getter function to fetch the uuid of the pool
+                Ok(PoolUuid::try_from(self.name.to_string())?)
+            }
+        }
+    }
+
+    /// get name of the pool from uuid
+    pub fn get_name_from_uuid(_uuid: &PoolUuid) -> PoolId {
+        unimplemented!()
+    }
+}
+
+impl<'de> Deserialize<'de> for ReplicaPool {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let json: Value = Value::deserialize(deserializer)?;
+        println!("json: {:?}", json);
+
+        let replica_pool = match json {
+            Value::Object(pool) => {
+                let name = match pool.get(&"name".to_string()) {
+                    Some(name) => match PoolId::try_from(name) {
```

Comment:
Can use instead the deser trait, example:
Some(name) => PoolId::deserialize(name).unwrap(),
But as pointed out above, I don't think we even need this if we move to an untagged. enum.

Reply:
Removed this custom deserializer.